test/src/ir_generation/tests/test_metadata.sw

```diff
@@ -0,0 +1,15 @@
+script;
+
+fn main() {}
+fn my_func() {}
+
+#[test()]
+fn my_test_func() {
+  my_func();
+}
+
+// check: fn main() -> (), $(main_md=$MD) {
+// check: fn my_test_func() -> (), $(test_md=$MD) {
+
+// check: $test_md = ($MD $MD)
```

Comment:
This could probably be a little more explicit, perhaps matching the `decl_id` tag is definitely referenced by the `test_md` index.